Mathlib/Analysis/Complex/Angle.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2024 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Analysis.SpecialFunctions.Complex.Arg
+import Mathlib.Geometry.Euclidean.Angle.Unoriented.Basic
+
+/-!
+# Angle between complex numbers
+
+This file relates the Euclidean geometric notion of angle between complex numbers to the argument of
+their quotient.
+
+## TODO
+
+Prove the corresponding results for oriented angles.
+
+## Tags
+
+arc-length, arc-distance
+-/
+
+open InnerProductGeometry
+open scoped Real
+
+namespace Complex
+variable {a x y : ℂ}
+
+/-- The angle between two non-zero complex numbers is the absolute value of the argument of their
+quotient.
+
+Note that this does not hold when `x` or `y` is `0` as the LHS is `π / 2` while the RHS is `0`. -/
+lemma angle_eq_abs_arg (hx : x ≠ 0) (hy : y ≠ 0) : angle x y = |(x / y).arg| := by
+  refine Real.arccos_eq_of_eq_cos (abs_nonneg _) (abs_arg_le_pi _) ?_
+  rw [Real.cos_abs , Complex.cos_arg (div_ne_zero hx hy)]
+  have := (map_ne_zero Complex.abs).2 hx
+  have := (map_ne_zero Complex.abs).2 hy
+  simp [div_eq_mul_inv, Complex.normSq_eq_norm_sq]
```

Comment:
I was told nonterminal `simp`s are frowned above, as opposed to `simp only`s generated by `simp?`.  This case seems reasonable though given the robustness of the remaining tactics.

Reply:
Yes, by "nonterminal simp" we mean "simp before a rigid tactic like `exact`, `apply` or `refine`". But `field_simp` and `ring` are definitely not rigid.